arviz/utils.py

```diff
@@ -753,3 +767,95 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
 
     return wrapper
+
+def Calculate_ICS(
+    dataset_dict,
+    scale: Optional[ScaleKeyword] = None,
+    ic: Optional[ICKeyword] = None
+):
+
+    """
+    Helper Function used to execute the loo and waic information criteria
+
+    LOO is leave-one-out (PSIS-LOO `loo`) cross-validation and
+    WAIC is the widely applicable information criterion.
+    Read more theory here - in a paper by some of the leading authorities
+    on model selection dx.doi.org/10.1111/1467-9868.00353
+
+    Parameters
+    ----------
+
+    dataset_dict :  dataset_dict: dict[str] -> InferenceData or ELPDData
+    A dictionary of model names and InferenceData or ELPDData objects
+    scale: str, optional
+        Output scale for IC. Available options are:
+
+        - `log` : (default) log-score (after Vehtari et al. (2017))
+        - `negative_log` : -1 * (log-score)
+        - `deviance` : -2 * (log-score)
+
+        A higher log-score (or a lower deviance) indicates a model with better predictive
+        accuracy.
+    ic: str, optional
+        Information Criterion (PSIS-LOO `loo` or WAIC `waic`) used to compare models. Defaults to
+        ``rcParams["stats.information_criterion"]``.
+
+    Returns
+    -------
+
+    [WIP]
+
+    """
+    names = list(dataset_dict.keys())
+
+    if scale is not None:
+        scale = cast(ScaleKeyword, scale.lower())
+    else:
+        scale = cast(ScaleKeyword, rcParams["stats.ic_scale"])
+    allowable = ["log", "negative_log", "deviance"] if NO_GET_ARGS else get_args(ScaleKeyword)
+    if scale not in allowable:
+        raise ValueError(f"{scale} is not a valid value for scale: must be in {allowable}")
+    if scale == "log":
+            scale_value = 1
+            ascending = False
+    else:
+        if scale == "negative_log":
+            scale_value = -1
+        else:
+            scale_value = -2
+        ascending = True
+
+
+    if ic is None:
+        ic = cast(ICKeyword, rcParams["stats.information_criterion"])
+    else:
+        ic = cast(ICKeyword, ic.lower())
+    allowable = ["loo", "waic"] if NO_GET_ARGS else get_args(ICKeyword)
+    if ic not in allowable:
+        raise ValueError(f"{ic} is not a valid value for ic: must be in {allowable}")
+
+    # I have to return loo or waic in order for compare to create the df_comp and scale col
+
+    if ic == "loo":
+        ic_func: Callable = loo
+    elif ic == "waic":
+        ic_func = waic
+    else:
+        raise NotImplementedError(f"The information criterion {ic} is not supported.")
+    
+
+    names = []
+    dataset_dict = deepcopy(dataset_dict)
+    for name, dataset in dataset_dict.items():
+        names.append(name)
+        if not isinstance(dataset, ELPDData):
+            try:
+                dataset_dict[name] = ic_func(convert_to_inference_data(dataset), pointwise=True, scale=scale)
```

Comment:
I think this is missing the `var_name` argument that can be used in `loo` and `waic` and should also be available in compare and plot_elpd.